src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage_FlushWrites.cs

```diff
@@ -71,12 +71,22 @@ private void FlushInMemoryDataToDisk()
 
             using var _ = _connectionPool.Target.GetPooledConnection(out var connection);
 
-            connection.RunInTransaction(static state =>
+            var exception = connection.RunInTransaction(static state =>
             {
                 state.self._solutionAccessor.FlushInMemoryDataToDisk_MustRunInTransaction(state.connection);
                 state.self._projectAccessor.FlushInMemoryDataToDisk_MustRunInTransaction(state.connection);
                 state.self._documentAccessor.FlushInMemoryDataToDisk_MustRunInTransaction(state.connection);
-            }, (self: this, connection));
+            },
+            (self: this, connection),
+            throwOnSqlException: false);
+
+            if (exception != null)
+            {
+                // Some sql exception occurred (like SQLITE_FULL). These are not exceptions we can suitably recover
+                // from.  In this case, transition the storage instance into being unusable. Future reads/writes will
+                // get empty results.
+                this.DisableStorage();
+            }
```

Comment:
any failure when flushing our memory to disk results in us being shut down.  This is important as we don't want to continue trying to hit the disk over and over again, or continually buffer more data into memory hoping the disk will eventually become available.